dedup/src/main.rs

```diff
@@ -0,0 +1,119 @@
+use std::fs::File;
+use std::path::{Path,PathBuf};
+use std::collections::HashMap;
+use std::collections::hash_map::Entry;
+use std::io::{self, Read};
+
+extern crate md5;
+extern crate walkdir;
+
+const FOLLOW_SYMLINKS: bool = false;
+const ROOT_DIRECTORY: &'static str = "/";
+//const ROOT_DIRECTORY: &'static str = "/home/owen";
+//const REPORT_ERRORS: bool = false;
+const BUFFER_SIZE: usize = 8096;   // 20MB  // TODO: tweak
+
+type FileIter = Box<Iterator<Item=walkdir::DirEntry>>;
+
+/*
+enum SelectionStrat {
+    PathLen,
+    CreationTime,
+    UpdateTime,
+    //Drive(String),
+    Prefix(String),
+    NotPrefix(String),
+}
+type SelectionStrats = Vec<SelectionStrat>;
+
+enum ResolutionStrat {
+    Nothing,
+    Delete,
+    Symlink,
+}
+*/
+
+fn get_info(path: &Path) -> io::Result<(u64,[u8;16])> {
+    let mut f = File::open(path)?;
+    let mut buffer: Box<[u8]> = vec![0u8; BUFFER_SIZE].into_boxed_slice();
+    let mut total = 0usize;
+    let mut context = md5::Context::new();
+    loop {
+        let size = f.read(&mut buffer)?;
+        if size == 0 {
+            let digest = context.compute();
+            return Ok((total as u64, *digest));
```

Comment:
Usually you omit return statements but I think a return here made more sense because of the loop. Any keyword that breaks/jumps/whatever (e.g. `break`, `return`, `continue`) can be used without a semicolon but using it is not a warning/error